tools/templates/moar/Makefile.in

```diff
@@ -106,7 +106,15 @@ m-install: m-all
 	$(CP) $(NQP_MOAR) @q($(DESTDIR)$(NQP_LIB_DIR))@
 	$(CP) $(P5QREGEX_MOAR) @q($(DESTDIR)$(NQP_LIB_DIR))@
 	$(CP) @nfp(src/vm/moar/profiler/template.html)@ @nfpq($(DESTDIR)$(NQP_LIB_DIR)/profiler/template.html)@
-	$(PERL5) @shquot(@script(install-moar-runner.pl)@)@ "$(DESTDIR)" @q($(PREFIX))@ @q($(NQP_LIB_DIR))@ @nfpq(@moar@)@
+@perl(
+	if ( $config{relocatable} ) {
+		$out .= $macros->expand('	$(PERL5) @shquot(@configure_script@)@ @configure_opts()@ --set-var=nqplibdir=@shquot(@nqplibdir@)@ --expand=@nfpq(@templates_dir@/@platform@/nqp-m-reloc)@ --out=@q($(DESTDIR)$(BIN_DIR)/$(M_RUNNER))@');
+	}
+	else {
+		$out .= $macros->expand('	$(PERL5) @shquot(@configure_script@)@ @configure_opts()@ --set-var=nqplibdir=@shquot(@nqplibdir@)@ --expand=@nfpq(@templates_dir@/@platform@/nqp-m)@ --out=@q($(DESTDIR)$(BIN_DIR)/$(M_RUNNER))@');
+	}
+)@
```

Comment:
In this block you only use perl for setting the right template name of `nqp-m`. Much tidier would it be to set a config variable like `reloc_suffix` and then make it:

`@nfpq(@templates_dir@/@platform@/nqp-m@reloc_suffix@)@`

BTW, when I used `@templates_dir@` it was to work around a bug, if memory serves me right. `@template(@platform@/nqp-m)@` works now. `@shquot()@` or `@q()@` would do the quoting.

Reply:
Changed as proposed. Thanks for the idea!